grants/2022_CZI_EOSS5/full_application.md

```diff
@@ -0,0 +1,197 @@
+# scikit-image EOSS5 full application
+
+# Proposal details
+
+## 0. 
+
+## 1. Proposal title: From library to protocol: scikit-image as an API reference
+
+## 2. Previous funding
+
+Did you previously apply for funding for this or a related proposal under the CZI EOSS program?
+
+- Yes
+
+## 3. Proposal purpose
+
+#### Describe the purpose of the proposal in one sentence (maximum 200 characters):
+
+- *Example: To develop a comprehensive, validated atlas of the human kidney at single-cell resolution open to the entire scientific and clinical community.*
+
+To create a consistent, typed-annotated, discoverable, and extensible API for scikit-image and facilitate interoperability in the broader image analysis ecosystem.
+
+## 4. Amount requested
+
+$200K/y for 2y
+
+## 5. Proposal summary
+
+*Provide a short summary of the application (maximum of 500 words) (auto-filled from LOI; update if needed)*
+
+Scikit-image is among the most popular image processing frameworks for scientific image analysis. This proposal aims to support its contributors and maintainers with
+- a formalization and typing of scikit-image's API;
+- ongoing maintenance and community-driven improvements; and
+- mentoring and supporting diverse contributors.
+
+### New typed API and protocols
+
+Scikit-image's popularity can be attributed to its first-class support for n-dimensional NumPy arrays. This support gave scikit-image seamless interoperability with the core scientific Python libraries of not just NumPy itself, but also SciPy, scikit-learn, networkx, and others, enabling users to rapidly create custom analyses drawing from many different scientific libraries.
+
+However, specialized array formats have proliferated for specific needs such as deep learning (PyTorch, TensorFlow), general purpose GPU computing (CuPy, PyOpenCL), and out-of-core and distributed computing (Dask, zarr). The communities that grew around each of these formats came together to define a common [Array API] (application programming interface), allowing users to easily transition between array formats without having to rewrite code.
+
+This proposal aims to do the same for image analysis libraries.
+
+Thanks to its popular position in the ecosystem, scikit-image is already seen as a "standard" API for the community to emulate. For example, NVIDIA's cuCIM library provides a `cucim.skimage` module that emulates the scikit-image API for CuPy arrays. However, since scikit-image grew organically from the contributions of a broad community -- almost 500 direct code contributors -- API inconsistencies exist. Rather than perpetuate these throughout a broader ecosystem, we want to (1) formalize protocols and types for image analysis, and (2) reorganize and annotate scikit-image’s API accordingly. With this, we aim to provide a reference implementation that can be easily extended by other libraries, thereby benefiting usability, discoverability, performance, and community innovation.
```

Comment:
```suggestion
Thanks to its popular position in the ecosystem, scikit-image is already seen as a "standard" API for the community to emulate. For example, NVIDIA's cuCIM library provides a `cucim.skimage` module that emulates the scikit-image API for CuPy arrays. However, since scikit-image grew organically from the contributions of a broad community -- almost 500 direct code contributors -- API inconsistencies exist. Rather than perpetuate these throughout a broader ecosystem, we want to (1) formalize protocols and types for image analysis, and (2) reorganize and annotate scikit-image's API accordingly. With this, we aim to provide a reference implementation that can be easily extended by other libraries, thereby benefiting usability, discoverability, performance, and community innovation.
```

(I actually prefer these 'real' apostrophes, but here it's different from the rest.)